### PR TITLE
plugins/treesitter-textobjects: init

### LINKED ISSUE #722: nvim-treesitter-textobjects
**Plugin**: What's the name of the plugin
nvim-treesitter-textobjects
**Plugin repository**: URL to the repository of the plugin
https://github.com/nvim-treesitter/nvim-treesitter-textobjects
**Is the plugin on nixpkgs**: Is this plugin available on nixpkgs?
[yes](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=treesitter-textobjects)
## Extra information
Please provide extra information about the plugin, and any ideas you might have regarding nixvim-specific options.
Allows easy selection/movement/swapping of different text objects such as functions, parameters, classes.

Would be nice to have different subcategories for move/swap/select and lsp_interop, probably with keymaps as an option.

Select has a lookahead option to jump forward, Move has an additional capability regarding using queries/query groups but to be honest I don't understand it well enough to explain.

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for [treesitter-textobjects](https://github.com/nvim-treesitter/nvim-treesitter-textobjects).

Fixes #722.

### CODE REVIEW

**Comment by traxys** [comment]
Disk full...
```
error: executing SQLite statement 'insert or replace into NARs(cache, hashPart, timestamp, present) values (2, 'bx1bx1fmp49j356lgcqy3zk66wddvarn', 1700736613, 0)': database or disk is full, database or disk is full (in '/root/.cache/nix/binary-cache-v6.sqlite')
```

File: plugins/languages/treesitter/treesitter-textobjects.nix [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+          Same as `gotoNextStart`, but it jumps to the start of the text object.
+        '';
+
+        gotoPreviousStart = mkKeymapsOption ''
+          Same as `gotoNextStart`, but it jumps to the previous text object.
+        '';
+
+        gotoPreviousEnd = mkKeymapsOption ''
+          Same as `gotoNextEnd`, but it jumps to the previous text object.
+        '';
```
Comment (traxys) [inline plugins/languages/treesitter/treesitter-textobjects.nix:154]: Isn't this laking:
```lua
      -- Below will go to either the start or the end, whichever is closer.
      -- Use if you want more granular movements
      -- Make it even more gradual by adding multiple queries and regex.
      goto_next = {
        ["]d"] = "@conditional.outer",
      },
      goto_previous = {
        ["[d"] = "@conditional.outer",
      }
```

File: plugins/languages/treesitter/treesitter-textobjects.nix [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+      };
+
+      lspInterop = {
+        enable = helpers.defaultNullOpts.mkBool false "LSP interop.";
+
```
Comment (traxys) [inline plugins/languages/treesitter/treesitter-textobjects.nix:159]: Insn't this lacking:
```lua
      border = 'none',
```

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/languages/treesitter/treesitter-textobjects.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/languages/treesitter/treesitter-textobjects.nix:154]: Indeed, nice catch.
Fixed.

**Review by traxys — APPROVED** [review]

**Pull request merged**